builtin/game/item_entity.lua

```diff
@@ -307,16 +337,11 @@ core.register_entity(":__builtin:item", {
 		end
 	end,
 
-	on_punch = function(self, hitter)
-		local inv = hitter:get_inventory()
-		if inv and self.itemstring ~= "" then
-			local left = inv:add_item("main", self.itemstring)
-			if left and not left:is_empty() then
-				self:set_item(left)
-				return
-			end
+	on_punch = function(self, hitter, ...)
```

Comment:
I think it should be up to whoever decides to expand on_punch to update on_pickup, as they're strictly intertwined. I'd avoid the `...`, I personally find it more confusing than helping

Reply:
This might, however be impossible or very hard in the future.